tests/parser/functions/test_rlp_list.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+from tests.setup_transaction_tests import chain as s, tester as t, ethereum_utils as u, check_gas, \
+    get_contract_with_gas_estimation
+
+def test_rlp_decoder_code():
+    import rlp
+    rlp_decoder_code = """
+u: bytes <= 100
+
+def foo() -> address:
+    x = RLPList('\xf6\x9455555555555555555555\xa0GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG', [address, bytes32])
+    return x[0]
+
+def fop() -> bytes32:
+    x = RLPList('\xf6\x9455555555555555555555\xa0GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG', [address, bytes32])
+    return x[1]
+
+def foq() -> bytes <= 100:
+    x = RLPList('\xc5\x83cow\x03', [bytes, num])
+    return x[0]
+
+def fos() -> num:
+    x = RLPList('\xc5\x83cow\x03', [bytes, num])
+    return x[1]
```

Comment:
@DavidKnott this is the one scenario where we actually support mixed types in a list ;)

Reply:
@jacqueswww Yup, we should probably add a nice section on RLPList to the docs, and then link to the Ethereum wiki.